### ISSUE TITLE
Enforce sharing of email address if it is disabled on SignIn with Facebook

### ISSUE THREAD

**Issue by avkonst** [firebase/firebaseui-web#528]
**Is your feature request related to a problem? Please describe.**
Sign In with Facebook allows a user to opt-out from sharing an email address. As a result user's info received by the Firebase does not have an email address. Why is it a problem?:

- Some applications may rely on the email address being always available (my use case). As a workaround, I need to handle the case post-signup and ask a user to provide the email and forward it to my backend service. This is OK if you have got such a backend. And even if there is such a backend, the Firebase Authentications list will still miss the email address for a user.
- Later, the same user may signin using google provider or email and the same account will not be detected, i.e. Facebook account without email is not matched by the dialog. The result is 2 different accounts for the same user and bad user's experience. I have not found a solution to this problem.

**Describe the solution you'd like**
It would be great to have signin dialog to dispatch the case of missed email address and ask a user to provide it before the flow is completed (similarly to `sign in by email`, the dialog asks for full user's name after signup)

This is where user can opt-out from sharing the email address:
![image](https://user-images.githubusercontent.com/3971413/48820035-dac5e300-edb7-11e8-81df-86546791b228.png)

**Comment by bojeil-google** [comment]
Hey @avkonst, just to confirm, you are requesting the option to add an email to an account if the user signs in with an identity provider that didn't return an email for the user.
Note, for now, after sign-in completion with FirebaseUI, you can always show a page to ask the user to provide their email if the user doesn't have one. You would just call `user.updateEmail(email)` after they provide the email.

**Comment by avkonst** [comment]
Yes, this my current workaround I am thinking to add. I assume the email
provided via updateEmail will be saved as not verified. Will existing
firebase email verification procedure still work after I invoke updateEmail
and trigger verification?

**Comment by bojeil-google** [comment]
After you call `user.updateEmail`, trigger `user.sendEmailVerification()`. It will send a verification email to the user's email.

**Comment by garyforsterio** [comment]
:+1:

**Comment by dackers86** [comment]
Thank you for taking the time to submit this request

This is a great idea and definitely something we can consider for the future. It is not currently prioritised for our immediate new version roadmap, as we are focusing on a new upcoming release.

In the meantime, we will keep this issue open so we can revisit it during a later planning cycle, particularly as we gather more feedback from the community.
